lib/hawkular/inventory/inventory_api.rb

```diff
@@ -132,11 +138,11 @@ def get_config_data_for_resource(resource_id, feed)
     # @return [Array<Resource>] List of resources that are children of the given parent resource.
     #   Can be empty
     def list_child_resources(parent_resource, recursive = false)
```

Comment:
Shouldn't we go the full step here and require the canonical path `/t../f../r..` ?
I recall that I talked with @abonas some time ago that very often the actual objects are thrown away and only the "primary key" is retained. Users would still get the full object back from where they can `Resource.path` to get the path. 
While we are here, we could even make `Resource.id` the canonical path (primary key) and retain the original ìd` from Hawkular-Inventory under some different name.